processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -214,7 +214,26 @@ public File mergeQueryableIndex(
     return mergeQueryableIndex(indexes, rollup, metricAggs, outDir, indexSpec, new BaseProgressIndicator());
   }
 
-  public File mergeQueryableIndex(
+  public File mergeAndCloseMergee(
+      List<QueryableIndex> indexes,
+      boolean rollup,
+      final AggregatorFactory[] metricAggs,
+      File outDir,
+      IndexSpec indexSpec,
+      ProgressIndicator progress
+  ) throws IOException
+  {
+    try {
+      return mergeQueryableIndex(indexes, rollup, metricAggs, outDir, indexSpec, progress);
+    }
+    finally {
```

Comment:
Looks like some call sites such as unit tests call the no-progress-indicator version and close the subindexes themselves, those may need adjusting, if we go with both method closing the subindexes. I think it's fine if that's the approach taken and that makes this PR bigger.